web/apps/labelstudio/src/pages/Settings/AnnotationSettings/ModelVersionSelector.js

```diff
@@ -0,0 +1,97 @@
+import { useCallback, useContext, useEffect, useState } from 'react';
+import { useAPI } from '../../../providers/ApiProvider';
+import { Button } from '../../../components';
+import { Form, Label, Select } from '../../../components/Form';
+import { ProjectContext } from '../../../providers/ProjectProvider';
+
+
+export const ModelVersionSelector = ({
+  name = "model_version",
+  valueName = "model_version",
+  apiName = "projectModelVersions",
+  ...props
+}) => {
+  const api = useAPI();
+  const { project, updateProject } = useContext(ProjectContext);
+  const [loading, setLoading] = useState(true);
+  const [versions, setVersions] = useState([]);
+  const [models, setModels] = useState([]);
+  const [version, setVersion] = useState(null);
+  const [placeholder, setPlaceholder] = useState("");
+  
+  useEffect(() => {
+    setVersion(project?.[valueName] || null);
+  }, [project?.[valueName], versions]);
+
+  const fetchMLVersions = useCallback(async () => {
+    const pk = project?.id;
+
+    if (!pk) return;
+
+    const modelVersions = await api.callApi(apiName, {
+      params: {
+        pk: pk,
+        extended: true,
+        include_live_models: true
+      },
+    });
+
+    if (modelVersions) {
+      if (modelVersions.live.length > 0) {
+        
+        const liveModels = modelVersions.live.map(item => {
+          const label = item.title + " (" + item.readable_state + ")";
+          return {
+            group: "Models",
+            value: item.title,
+            label: label
+          };
+        });
+        
+        setModels(liveModels);
+      }
+
+      if (modelVersions.static.length > 0) {
+        const staticModels = modelVersions.static.map(item => {
+          const label = item.model_version + " (" + item.count + " predictions)";
+          return {
+            group: "Predictions",
+            value: item.model_version,
+            label: label
+          };
+        });
+        
+        setVersions(staticModels);
+      }
+
+      if (!(modelVersions.static?.length) && !(modelVersions.live?.length)) {
+        setPlaceholder("No model or predictions available");
+      }      
+    }
```

Comment:
Can use null safe checks to remove the additional wrapping if statement:

```suggestion
    if (modelVersions?.live?.length > 0) {
      const liveModels = modelVersions.live.map(item => {
        const label = item.title + " (" + item.readable_state + ")";
        return {
          group: "Models",
          value: item.title,
          label: label
        };
      });
      
      setModels(liveModels);
    }

    if (modelVersions?.static?.length > 0) {
      const staticModels = modelVersions.static.map(item => {
        const label = item.model_version + " (" + item.count + " predictions)";
        return {
          group: "Predictions",
          value: item.model_version,
          label: label
        };
      });
      
      setVersions(staticModels);
    }
  
    if (!(modelVersions?.static?.length) && !(modelVersions.live?.length)) {
      setPlaceholder("No model or predictions available");
    }      
```